src/self-managed-debug/src/main.rs

```diff
@@ -0,0 +1,348 @@
+            } else {
+                eprintln!("Failed to export {}: {}", &previous_logs_file_name, e);
+            }
+        }
+
+        if let Err(e) = async {
+            let logs = pods
+                .logs(
+                    &pod_name,
+                    &LogParams {
+                        timestamps: true,
+                        ..Default::default()
+                    },
+                )
+                .await?;
+
+            if logs.is_empty() {
+                println!("No logs found for pod {}", pod_name);
+                return Ok(());
+            }
+            let mut file = File::create(&current_logs_file_name)?;
+
+            file.write_all(logs.as_bytes())?;
+
+            println!("Exported {}", &current_logs_file_name);
+            Ok::<(), anyhow::Error>(())
+        }
+        .await
+        {
+            eprintln!("Failed to export {}: {}", &current_logs_file_name, e);
+        }
+    }
+    Ok(())
+}
+
+/// Write k8s events to a yaml file.
+async fn dump_k8s_events(
+    context: &Context,
+    client: Client,
+    namespace: &String,
+) -> Result<(), anyhow::Error> {
+    let events: Api<Event> = Api::<Event>::namespaced(client.clone(), namespace);
+
+    let event_list = events.list(&ListParams::default()).await?;
+
+    if event_list.items.is_empty() {
+        println!("No events found for namespace {}", namespace);
+        return Ok(());
+    }
+
+    let file_path = format_resource_path(context.start_time, namespace, &K8sResourceType::Event);
+    let file_name = format!("{}/events.yaml", file_path);
+    create_dir_all(&file_path)?;
+    let mut file = File::options().append(true).create(true).open(&file_name)?;
+
+    for event in event_list {
+        serde_yaml::to_writer(&mut file, &event)?;
+    }
+
+    println!("Exported {}", &file_name);
+
+    Ok(())
+}
+
+/// Runs `kubectl describe` for a given resource and writes the output to a file.
+async fn dump_kubectl_describe(
+    context: &Context,
+    namespace: &String,
+    resource_type: &K8sResourceType,
+) -> Result<(), anyhow::Error> {
+    let resource_type_str = resource_type.to_string();
+    let mut args = vec!["describe", &resource_type_str, "-n", namespace];
+
+    if let Some(k8s_context) = &context.args.k8s_context {
+        args.extend(["--context", k8s_context]);
+    }
+
+    let output = tokio::process::Command::new("kubectl")
+        .args(args)
+        .stderr(std::process::Stdio::null()) // Silence stderr
+        .output()
+        .await?;
+
+    if !output.status.success() {
+        return Err(anyhow::anyhow!(
+            "{}",
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    if output.stdout.is_empty() {
+        println!("No {} found in namespace {}", resource_type, namespace);
+        return Ok(());
+    }
+
+    let file_path = format_resource_path(context.start_time, namespace, resource_type);
+    let file_name = format!("{}/describe.txt", file_path);
```

Comment:
It's not super important, but for cross-platform stuff, you should probably use PathBuf and Path instead of formatting Strings.